scripts/west_commands/build.py

```diff
@@ -155,7 +155,7 @@ def do_run(self, args, remainder):
                     self.run_cmake = True
         else:
             self.run_cmake = True
-        self._setup_source_dir()
+        self.source_dir = self._find_source_dir()
```

Comment:
I don't understand the need for this change. The function currently sets self.source_dir to what it finds. But it's being changed to return a directory, which is always being assigned to self.source_dir.

Reply:
I didn't want to set `self.source_dir` when setting up the build folder. In that case the result is assigned to a local variable, not to a class member.

Reply:
Crap, I saw that and meant to delete this comment before posting the review; sorry.